### DISCUSSION TITLE
Model Scaling Clarification

### DISCUSSION THREAD

**Discussion by ghost** [WEC-Sim/WEC-Sim#1364] [Q&A] [answered]
**The development team requests that before submitting a question on the theory and implementation behind WEC-Sim to first please:**
1. Read through our [WEC-Sim Documentation](http://wec-sim.github.io/WEC-Sim/man/theory.html) with additional description found in our [Advanced Features Section](http://wec-sim.github.io/WEC-Sim/man/advanced_features.html)
1. Review MATLAB source code ([functions](https://github.com/WEC-Sim/WEC-Sim/tree/main/source/functions), [objects](https://github.com/WEC-Sim/WEC-Sim/tree/main/source/objects)) and the Simulink blocks ([libraries](https://github.com/WEC-Sim/WEC-Sim/tree/main/source/lib))
1. Review the curated [WEC-Sim Examples](https://github.com/WEC-Sim/WEC-Sim/tree/main/examples) and the [WEC-Sim applications](https://github.com/WEC-Sim/WEC-Sim_Applications)
1. Review the [WEC-Sim issues page](https://github.com/WEC-Sim/WEC-Sim/issues) which allows you to search past resolved issues or questions that might help answer your own question.

**Describe the theory or implementation approach that you have a question**

Hello, I am new to using  WEC-SIM and  I'm having issues updating the existing WECCCOMP model to a larger scale. My goal is to increase the scale and input existing wave profiles  (eta file & spectrum file) to simulate real-life wave power generation. I have increased the size of all the geometry files in solidworks and ensured the coordinates are kept the same before exporting the new stl. 
When running the model, my float body is not correctly connected to the arm geometry (See screenshot) like in the smaller-scale version.  

I also noticed that when running the default regular waves, my arm movement rotates into the water. 
I have already updated the other necessary parameters using Froude scaling. 
[WaveStar_Scaled.zip](https://github.com/user-attachments/files/17612704/WaveStar_Scaled.zip)

**Describe the type of conclusion or resolution you are looking for**

My goal is just to simulate the power extraction of a larger-scale Wavestar device. 

Screen Shots:
![image](https://github.com/user-attachments/assets/7d6e8c9e-eb1e-4538-9cb3-c09e2e8c8d9b)

![image](https://github.com/user-attachments/assets/0b34ae60-fa77-40ae-8239-ae089d63e506)

**Comment by MShabara** [comment, marked as answer]
Hi @dhkgit 

When you scale the model, all the hydrodynamic coefficients change. This means you’ll need to generate a new .h5 file for your hydrodynamic body `body(1)` using a BEM solver.

The .stl files are primarily for visualization in your model. They’re only involved in calculations when non-linear forces are enabled, which doesn’t apply to your current setup.

Additionally, scaling affects how Simscape interprets body reference frames and rotations. For a detailed explanation, refer to [Nathan's response](https://github.com/WEC-Sim/WEC-Sim/issues/480#issuecomment-789115860).

To better understand the impact of rotation and scaling, try opening the original geometry files (before scaling) in a CAD software or [BEMRosetta](https://github.com/BEMRosetta/BEMRosetta/releases). You’ll notice that geometries aren’t aligned. By applying a translation with the same magnitude as the center of gravity locations specified in body instances/classes (after running the model examples model). The same thing should happen when you use your geometry files and c.g locations.

Let me know if you have any further questions!
